### PR TITLE
Fix lint in Xcode 7.2, it requires `-destination`

### PR DESCRIPTION
**Pull request by neonichu**
While `watchos` and `tvos` are fairly straightforward, I chose iPhone 4s as destination for `ios` since it is seems like the most likely to work with any version of Xcode at this point.

### CODE REVIEW

**Comment by neonichu** [comment]
Doesn't seem to work with 7.1: ```xcodebuild: error: Unable to find a destination matching the provided destination specifier:
        { name:Apple Watch - 38mm }

    The requested device could not be found because multiple devices matched the request.

    Available destinations for the "App" scheme:
        { platform:watchOS Simulator, id:22836D46-E3AA-4420-B854-A51ECB70B92E, OS:2.0, name:Apple Watch - 38mm }```

**Comment by neonichu** [comment]
While this somewhat works, it is really brittle. I'll update the PR to use `xcrun simctl list` and IDs instead. There's the remaining problem of certain simulators not being present, but I guess that would have also failed previously?

**Comment by segiddins** [comment]
@neonichu not sure why a simulator needs to be present since we're only building? Maybe @samdmarshall can shed some light on this one?

**Comment by samdmarshall** [comment]
I'm pretty sure it is because it needs to resolve the Sim SDK properly -- which it can only do based on target device.

**Comment by neonichu** [comment]
I think the same. One possibility could be creating sims ourselves and tearing them down again, but I'd say we'll see if this turns out to be a problem in practice before we do that.

**Comment by neonichu** [comment]
Alright, this should be somewhat reliable now and will show a more readable error if a certain sim is not present.

**Comment by neonichu** [comment]
Hm, not sure about https://travis-ci.org/CocoaPods/CocoaPods/jobs/97435296#L2602

**Comment by neonichu** [comment]
¯\\\_(ツ)\_/¯ about the `xcodebuild` error, an "iPhone 4s" device appears to exist: <https://travis-ci.org/#L128>

**Comment by neonichu** [comment]
Probably has something to do with the custom runtimes which are installed on Travis?

**Comment by neonichu** [comment]
OK, I have figured out the issue — the "App" target generated as part of lint uses a deployment target different from the Pod being linted, will fix that in a separate PR. 

But also the algorithm of finding a suitable simulator in insufficient, as it doesn't take the deployment target into account, potentially choosing a simulator which is too old.

**Comment by neonichu** [comment]
Turns out the separate PR won't be necessary, the deployment target of the Pod **is** taken into account, but linting for frameworks correctly enforces 8.0 as a minimum.

File: CHANGELOG.md [hunk 1]
```diff
@@ -183,6 +183,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Boris Bügling](https://github.com/neonichu)
   [Samuel Giddins](https://github.com/segiddins)
   [#4550](https://github.com/CocoaPods/CocoaPods/issues/4550)
+* Fix lint in Xcode 7.2, it requires `-destination`.
```
Comment (segiddins) [inline CHANGELOG.md]: missing preceding newline

File: lib/cocoapods/validator.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 require 'active_support/core_ext/array'
 require 'active_support/core_ext/string/inflections'
+require 'fourflusher'
```
Comment (segiddins) [inline lib/cocoapods/validator.rb]: please make this an Autoload

File: lib/cocoapods/validator.rb [hunk 3]
```diff
@@ -705,11 +706,11 @@ def xcodebuild
       command = %w(clean build -workspace App.xcworkspace -scheme App -configuration Release)
       case consumer.platform_name
       when :ios
-        command += %w(CODE_SIGN_IDENTITY=- -sdk iphonesimulator)
+        command += %w(CODE_SIGN_IDENTITY=- -sdk iphonesimulator) + Fourflusher::SimControl.new.destination('iPhone 4s', deployment_target)
```
Comment (segiddins) [inline lib/cocoapods/validator.rb]: `+=` the fouflusher stuff on a newline?

File: spec/unit/validator_spec.rb [hunk 4]
```diff
@@ -428,18 +429,20 @@ def podspec_path
...
+        Executable.stubs(:which).with(:xcrun)
         Executable.expects(:which).with('xcodebuild').times(4).returns('/usr/bin/xcodebuild')
         command = %w(clean build -workspace App.xcworkspace -scheme App -configuration Release)
         Executable.expects(:capture_command).with('xcodebuild', command, :capture => :merge).once.returns(['', stub(:success? => true)])
-        Executable.expects(:capture_command).with('xcodebuild', command + %w(CODE_SIGN_IDENTITY=- -sdk appletvsimulator), :capture => :merge).once.returns(['', stub(:success? => true)])
-        Executable.expects(:capture_command).with('xcodebuild', command + %w(CODE_SIGN_IDENTITY=- -sdk iphonesimulator), :capture => :merge).once.returns(['', stub(:success? => true)])
-        Executable.expects(:capture_command).with('xcodebuild', command + %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator), :capture => :merge).once.returns(['', stub(:success? => true)])
+        Executable.expects(:capture_command).with('xcodebuild', command + %w(CODE_SIGN_IDENTITY=- -sdk appletvsimulator) + Fourflusher::SimControl.new.destination('Apple TV 1080p'), :capture => :merge).once.returns(['', stub(:success? => true)])
```
Comment (segiddins) [inline spec/unit/validator_spec.rb]: these lines are too long

**Comment by segiddins** [comment]
@neonichu :+1: once this passes on CI!

**Comment by neonichu** [comment]
Confused about the failure <https://travis-ci.org/CocoaPods/CocoaPods/jobs/99489322#L1729>

**Comment by segiddins** [comment]
@neonichu don't worry about it. I think it's because the fixture its linting has no watchOS support?

**Comment by segiddins** [comment]
@neonichu think I fixed it. will merge if CI passes

**Comment by orta** [comment]
and the world waits on the PR

**Comment by orta** [comment]
:green_book:

**Pull request merged**